openmdao/core/group.py

```diff
@@ -1270,17 +1271,25 @@ def _solve_nonlinear(self):
         # Execute guess_nonlinear if specified.
         # We need to call this early enough so that any solver that needs initial guesses has
         # them.
-        # TODO: It is pointless to run this ahead of non-iterative solvers.
-        for sub in self.system_iter(recurse=True):
-            if hasattr(sub, 'guess_nonlinear'):
-                with sub._unscaled_context(outputs=[sub._outputs], residuals=[sub._residuals]):
-                    sub.guess_nonlinear(sub._inputs, sub._outputs, sub._residuals)
+        # TODO: It is pointless to run this ahead of non-iterative solvers, but multi-level
+        # models seem to ned it.
+        self._guess_nonlinear()
```

Comment:
Does this mean all groups run a guess_nonlinear iteration at the start of each solve_nonlinear?